### PR TITLE
Draw box/violin pts as legend item when other parts have opacity 0

### PR DESCRIPTION
**Pull request by etpinard**
resolves https://github.com/plotly/plotly.js/issues/3836

before: https://codepen.io/etpinard/pen/pmjGEL?editors=0010
after: https://codepen.io/MojtabaSamimi/pen/JqYQjG?editors=0010 

cc @plotly/plotly_js @nicolaskruchten

### CODE REVIEW

File: src/components/legend/style.js [hunk 1]
```diff
@@ -356,12 +356,27 @@ module.exports = function style(s, gd) {
 
         pts.each(function() {
             var p = d3.select(this);
-            var w = boundLineWidth(undefined, trace.line, MAX_MARKER_LINE_WIDTH, CST_MARKER_LINE_WIDTH);
 
-            p.style('stroke-width', w + 'px')
-                .call(Color.fill, trace.fillcolor);
+            if((trace.boxpoints === 'all' || trace.points === 'all') &&
```
Comment (etpinard) [inline src/components/legend/style.js]: @nicolaskruchten do you use other `(box)points` values for your `px` strip traces?

If so, we might need to add logic for `marker.outliercolor`, `marker.line.outliercolor` and `marker.line.outlierwidth` in the block below.

Comment (alexcjohnson) [reply src/components/legend/style.js]: might be interesting to add a mode where we show all points but style [suspected]outliers differently - but currently these other modes both hide the non-outlier points entirely - which seems like an awfully bad idea to combine with hiding the box/violin!

Comment (nicolaskruchten) [reply src/components/legend/style.js]: for `px` I was planning on using `'all'`, because of @alexcjohnson's comment :) so this is looking great to me :)

**Comment by archmoj** [comment]
[Codepen After](https://codepen.io/MojtabaSamimi/pen/JqYQjG?editors=0010).

**Comment by antoinerg** [comment]
Looks good to me :dancer:

**Pull request merged**